test/e2e/self_provisioned_elasticsearch_test.go

```diff
@@ -76,18 +80,56 @@ func (suite *SelfProvisionedTestSuite) AfterTest(suiteName, testName string) {
 
 func (suite *SelfProvisionedTestSuite) TestSelfProvisionedESSmokeTest() {
 	// create jaeger custom resource
-	exampleJaeger := getJaegerSimpleProd()
+	jaegerInstanceName := "simple-prod"
+	exampleJaeger := getJaegerSimpleProd(jaegerInstanceName)
 	err := fw.Client.Create(goctx.TODO(), exampleJaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
 	require.NoError(t, err, "Error deploying example Jaeger")
 	defer undeployJaegerInstance(exampleJaeger)
 
-	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, "simple-prod-collector", 1, retryInterval, timeout)
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName+"-collector", 1, retryInterval, timeout)
 	require.NoError(t, err, "Error waiting for collector deployment")
 
-	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, "simple-prod-query", 1, retryInterval, timeout)
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName+"-query", 1, retryInterval, timeout)
 	require.NoError(t, err, "Error waiting for query deployment")
 
-	ProductionSmokeTest("simple-prod")
+	ProductionSmokeTest(jaegerInstanceName)
+}
+
+func (suite *SelfProvisionedTestSuite) TestIncreasingReplicas() {
+	jaegerInstanceName := "simple-prod"
+	exampleJaeger := getJaegerSimpleProd(jaegerInstanceName)
+	err := fw.Client.Create(goctx.TODO(), exampleJaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	require.NoError(t, err, "Error deploying example Jaeger")
+	defer undeployJaegerInstance(exampleJaeger)
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName+"-collector", 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for collector deployment")
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName+"-query", 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for query deployment")
+
+	updateNodeCount := 2
+	changeNodeCount(jaegerInstanceName, updateNodeCount)
+	updatedJaegerInstance := getJaegerInstance(jaegerInstanceName, namespace)
+	require.EqualValues(t, updateNodeCount, updatedJaegerInstance.Spec.Storage.Elasticsearch.NodeCount)
+
+	// Make sure there are 2 ES deployments. Note: The deployment name is based on  "elasticsearch-cdm-" + namespace + jaegerInstanceName + "-1"
+	// with dashes in the namespace and  jaegerInstanceName removed
+	for i := 1; i <= updateNodeCount; i++ {
```

Comment:
Although testing an update of the ES config is useful - we primarily need to test an update on the Jaeger components themselves. So possibly also increase query and collector replicas?